### ISSUE TITLE
Add support to create maven plugins

### ISSUE THREAD

**Issue by jmini** [selesse/jxlint#12]
When you create a maven plugin for a linter created with jxlint, you have a lot in common. I am working on two example linters presented on [this page](https://jmini.github.io/jxlint-experiments/):

* `jxlint-impl` (I have copied the example you provided with some little modifications)
* `txtlinter` (a linter to check some `txt` files)

My goal is to use them during a maven build and I consider my first solution (using `org.codehaus.mojo:exec-maven-plugin`) is not realy user friendly. I am developing 2 small maven plugins for each linter: `jxlint-impl-maven-plugin` and `txtlinter-maven-plugin`.

I have tried to mutualize all the requested code in a project called `jxlint-maven-support`. In my opinion, this belongs to the jxlint project itself. The code is bound to the jxlint framework (mapping options from maven to jxlint, calling the jxlint framework…) and should be released with each new version of the framework, on the same time and if possible also on maven central.

I consider that `jxlint-maven-support` should be contributed to the jxlint project.

I know jxlint itself it build with gradle but this should not be a problem. Because the `jxlint-maven-support` project itselft is just a simple `jar` project (I am not familiar with gradle, but I hope it can be built without maven)

Let me know what you think.

**Comment by jmini** [comment]
I have pushed a first version of my `jxlint-maven-support`. The main class is [AbstractJxlintMojo](https://github.com/jmini/jxlint-experiments/blob/master/jxlint-maven-support/src/main/java/com/selesse/jxlint/maven/AbstractJxlintMojo.java).

For the moment I do not support as much as configuration possibility as Jxlint does with the command lines arguments in the Main class of each linter (for example rules include, exclude could be interesting)

I also want to write some tests (at least for the method that maps the maven configuration into a `com.selesse.jxlint.model.ProgramOptions` instance. See `AbstractJxlintMojo#createProgramOptions()`)

I am interested on your feedback.

**Comment by selesse** [comment]
This looks really cool!

I'd be more than happy to include this in the repository as a separate module (`jxlint-maven`, or `jxlint-maven-support` like you suggested). I'd been meaning to get around to writing a Gradle plugin for it, as well.

For a "1.0" release of this plugin (which would map to jxlint 2.0), it would be great to have all the options supported.

On a related note, are there things you think would make sense to include for a 2.0 version of jxlint? Offhand, I was thinking it would be nice to be able to add rules via annotations rather than extending `AbstractLintRules`. I've already setup some ground work for Java 8 support in 2.0 as well. Let me know! :)

**Comment by jmini** [comment]
For the 2.0 version:
I am still at the beginning of implementing a real use-case with jxlint (evaluation phase, I created the [jxlint-experiments](http://jmini.github.io/jxlint-experiments) for that purpose). I report all my findings as issue here.

Something I have on my road map is to be able to display the errors and warnings directly on Jenkins. They provide some tooling for that (the [analysis-core](https://plugins.jenkins.io/analysis-core) plugin) and my idea is to write an extension of that plugin that is capable to import the jxlint XML report. I did not look at it yet, but this probably means that I will need to read the XML produced by jxlint. Maybe this reader also belongs somehow in the jxlint framework (but it really depends on what Jenkins needs).

---

> it would be great to have all the options supported

Yes definitively.

I think I would make sense to reuse the last part of the methods:

* `ProgramOptions#getRuleListFromOptionString(String)`
* `ProgramOptions#getCategoryListFromOptionString(String)`

(the logic that verify if a rule or category exists, the code after the splitter)

> "1.0" release of this plugin (which would map to jxlint 2.0),

We need to align the versions of `jxlint-maven` with the version of `jxlint`, otherwise we will never know which version of the `jxlint-maven` goes with which version of `jxlint`.  I am working against the `1.7.2` version of jxlint, my version of `jxlint-maven-support` should be `1.7.2`.
